libiop/bcs/merkle_tree.tcc

```diff
@@ -474,54 +515,79 @@ bool merkle_tree<FieldT, hash_digest_type>::validate_set_membership_proof(
         std::swap(S, new_S);
     }
 
+    // Add the cap, including the root's direct children and the root.
+    // The only elements should be the cap (not including the root).
```

Comment:
Should the cap include the root, or not?

Reply:
Good question. I think the cap doesn't include the root. It's been such a long time since I wrote this code so I can only guess.